src/Data.mo

```diff
@@ -136,11 +140,21 @@ module {
 
     public func getMaxEligibleDeposit(threshold : Nat) : ?Entry<K> {
       for (((deposit, key), value) in state.depositsTree.entriesRev()) {
-        if (deposit <= threshold) return null;
+        if (deposit <= threshold) {
+          state.unusable_deposit.sum := state.deposit_sum;
+          state.unusable_deposit.correct := true;
+          return null;
+        };
         if (not value.lock) return ?Entry(true, key, value, state);
       };
       return null;
     };
+    
+    public func feeChanged() {
+      state.unusable_deposit.correct := false;
+    };
```

Comment:
So here we set this to false but what if it is true? Do we rely on activity (by the trigger timer) to call `getMaxEligibleDeposit` at which point it will get back to true?

Maybe we can do better and right away inspect the tip of the queue. Maybe we can just call `getMaxEligibleDeposit` here once?